Sources/Sentry/SentryFileManager.m

```diff
@@ -653,8 +653,11 @@ + (BOOL)createDirectoryAtPath:(NSString *)path withError:(NSError **)error
 
 - (void)createPathsWithOptions:(SentryOptions *_Nonnull)options
 {
-    NSString *cachePath
-        = NSSearchPathForDirectoriesInDomains(NSCachesDirectory, NSUserDomainMask, YES).firstObject;
+    NSString *cachePath = options.cacheDirectory;
+
+    if (cachePath == nil) {
+        cachePath = NSSearchPathForDirectoriesInDomains(NSCachesDirectory, NSUserDomainMask, YES).firstObject;
+    }
```

Comment:
`m`: I would limit setting the default value in the `SentryOptions` and remove this check here.